app/src/main/java/com/example/taskyy/ui/viewmodels/AgendaViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.taskyy.ui.viewmodels
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.taskyy.domain.usecases.LogoutUseCase
+import com.example.taskyy.ui.events.AgendaEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+import java.util.Locale
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val logoutUseCase: LogoutUseCase
+): ViewModel() {
+    var state by mutableStateOf(AgendaState())
+        private set
+
+    fun onEvent(event: AgendaEvent) {
+        when (event) {
+            is AgendaEvent.OnMonthExpanded -> state = state.copy(isMonthExpanded = event.isMonthExpanded)
+            is AgendaEvent.OnDateSelected -> formatSelectedDate(event.date)
+            is AgendaEvent.OnUserInitialsClicked -> state = state.copy(isUserDropDownExpanded = event.isUserDropDownExpanded)
+            is AgendaEvent.OnLogOutCLicked -> logout()
+        }
+    }
+
+    private fun formatSelectedDate(date: Long) {
+        val monthDate = SimpleDateFormat("MMMM", Locale.US)
+        val monthName = monthDate.format(date)
```

Comment:
Generally, I'd recommend to use `DateTimeFormatter` for date formatting. `SimpleDateFormat` is an older API which isn't thread-safe. In your example this won't lead to any issues, since your instance isn't shared across threads, but to build a habit, I'd start using the more modern API:
```
val now = LocalDateTime.now()
val formattedTime = DateTimeFormatter
    .ofPattern("dd MM yyyy")
    .format(now)
```
Generally, the `LocalDateTime` API is a lot more flexible and allows easy date time calculations